.vsts/templates/blazor-build.yml

```diff
@@ -1,31 +1,49 @@
-trigger:
-- master
-- release/*
+parameters:
+  queueName: ''
+  beforeBuild: []
+  afterBuild: []
+  variables:
+
 phases:
 - phase: Windows
   queue:
-    name: Hosted VS2017
+    name: ${{ parameters.queueName }}
     parallel: 2
     matrix:
-      Debug:
-        BuildConfiguration: Debug
       Release:
         BuildConfiguration: Release
+      Debug:
+        BuildConfiguration: Debug
+  variables:
+    ${{ insert }}: ${{ parameters.variables }}
   steps:
   - checkout: self
     clean: true
+
+  - task: NuGetCommand@2
+    displayName: Clear NuGet caches
+    inputs:
+      command: custom
+      arguments: 'locals all -clear'
+
+  - ${{ parameters.beforeBuild }}
+
   - script: .\build.cmd -ci /p:Configuration=$(BuildConfiguration)
     displayName: Run build.cmd
+
   - task: PublishTestResults@2
     displayName: Publish test results
     condition: always()
     inputs:
       testRunner: vstest
       testResultsFiles: 'artifacts/logs/**/*.trx'
+
   - task: PublishBuildArtifacts@1
     displayName: Upload artifacts
     condition: and(succeeded(), eq(variables['system.pullrequest.isfork'], false))
     inputs:
-      pathtoPublish: artifacts/
+      pathtoPublish: artifacts/$(BuildConfiguration)/
       artifactName: artifacts-$(BuildConfiguration)
       artifactType: Container
+
+  - ${{ parameters.afterBuild }}
```

Comment:
FYI I'm totally #YOLOing on these yaml files. Not only do I have a meh reaction to yml, I don't know what kinds of things they do in the context of VSTS.

Reply:
YAML docs: https://github.com/Microsoft/vsts-agent/blob/master/docs/preview/yamlgettingstarted.md

This `${{ }}` syntax is explained here: https://github.com/Microsoft/vsts-agent/blob/master/docs/preview/yamlgettingstarted-templateexpressions.md

Reply:
I don't want to know.. that would cause me to forget something else 😆